### DISCUSSION TITLE
Does mpremote have a test suite?

### DISCUSSION THREAD

**Discussion by TheChymera** [micropython/micropython#11308] [General]
I see [mpremote](https://github.com/micropython/micropython/tree/master/tools/mpremote) is its own package, and also distributed via pypi.
Does it not have a test suite? Or is it somewhere else?

**Comment by Josverl** [comment]
The short answer - no there is none yet - but I agree that it should be there.

In several changes I made I have created my own tests using pytest, but have not submitted them as part of the PR ,
perhaps we should pull  these and others together and add these to a new folder ( `tools\mpremote\tests`  or to `tests\mpremote` ) 
also - perhaps there should be 3 categories :
- pyboard.py 
- pyboardextended.py 
- mpremote  module 

For a test-suite: my preference is `pytest` as it requires much less boilerplate to create the tests, and it also works with any tests following the unittest format. 

@jimmo : any preferences / suggestions on how to assist ?. 

A [few of my tests are here ](https://github.com/Josverl/mpremote/tree/main/tests) for now: others I may have lost after removing branches after a PR, but I may be able to dig them up or re-write.  

Note that some tests may/will still require manual interaction, and therefore cannot be run in CI/CD, 
So I included detect for the connected devices ( serial only at this moment) and for the `port` (not serialport) of the device as I was tackling an esp32/8266 issue.
These can be flagged with `@pytest.mark.manual` and skipped in CI/CD  using `pytest -m "not manual"`, 
and run manually `pytest -m manual` to reproduce issues / validate fixes.
I use a timeout to continue if no action was confirmed (press spacebar) and then treat the test as skipped. 

![image](https://user-images.githubusercontent.com/981654/233784554-ff7daf82-ec7f-4cd9-a3e9-6234fb1f681e.png)
